Add lab1

miarstfirst opened the pull request:
первая лабораторная

miarstfirst committed:
Add lab1

AlekseiMaslakov commented on lab1/src/Main/Main.java:
```diff
@@ -0,0 +1,28 @@
+        hero.move();
+        hero.move();
+
+        hero.setMoving(new FlightMove());
```
По заданию, поведение героя выбирает пользователь

AlekseiMaslakov commented on lab1/src/Hero/Hero.java:
```diff
@@ -0,0 +1,14 @@
+import MovingType.*;
+
+public class Hero {
+    MovingType movingType;
```
Имеет смысл закрывать от внешнего мира всё что можно закрыть, в данном случае переменной ни к чему быть видной вовне - для работы уже есть методы setMoving и move

AlekseiMaslakov commented on lab1/src/Hero/Hero.java:
```diff
@@ -0,0 +1,14 @@
+
+    public void move() {
+        movingType.move();
+    }
```
Можно установить null в качестве movingType, что приведёт к ошибке при вызове метода move()

miarstfirst committed:
фикс зачений:
- введен метод play у класса Hero
- поле movingType класса Hero помечено как private
- введена проверка на null при вызове setMoving

AlekseiMaslakov commented on lab1/src/Main/Main.java:
```diff
@@ -0,0 +1,41 @@
+        System.out.println("Введите что-нибудь (для выхода введите 0):");
+
+        while (true) {
+            userInput = scanner.nextInt();
```
Если пользователь введёт не число - будет ошибка

miarstfirst committed:
фикс зачений:
- отбрасываем при вводе все кроме чисел
